lib/grape-swagger/endpoint.rb

```diff
@@ -288,7 +288,8 @@ def expose_params_from_model(model)
     end
 
     def model_name(name)
-      name.respond_to?(:name) ? name.name.demodulize.camelize : name.split('::').last
+      name = name.to_s.split('::')[0..-2].join('::').constantize if name.to_s.end_with?('Entity', 'Entities')
```

Comment:
good point … I did it, cause the name could be very long, see the spec, without it the name would be: `TestDefinitionClass1`